File: app/src/main/java/com/vsa/filmoteca/data/repository/MoviesPersistanceRepository.java
package com.vsa.filmoteca.data.repository;

import com.vsa.filmoteca.data.domain.Movie;
import com.vsa.filmoteca.data.repository.database.MoviesDataBaseSource;
import com.vsa.filmoteca.data.repository.sharedpreferences.SharedPreferencesManager;

import java.util.List;

/**
 * Created by albertovecinasanchez on 22/7/16.
 */

public class MoviesPersistanceRepository {

    private MoviesDataBaseSource mMoviesDataBaseSource;
    private SharedPreferencesManager mSharedPreferencesManager;

    public MoviesPersistanceRepository(MoviesDataBaseSource moviesDataBaseSource, SharedPreferencesManager sharedPreferencesManager) {
        mMoviesDataBaseSource = moviesDataBaseSource;
        mSharedPreferencesManager = sharedPreferencesManager;
    }

    public int getCurrentMovieIndex() {
        return mSharedPreferencesManager.getCurrentMovieIndex();
    }

    public void setCurrentMovieIndex(int index) {
        mSharedPreferencesManager.setCurrentMovieIndex(index);
    }

    public int getMoviesCount() {
        return mSharedPreferencesManager.getMoviesCount();
    }

    public void setMoviesCount(int size) {
        mSharedPreferencesManager.setMoviesCount(size);
    }

    public Movie getCurrentMovie() {
        mMoviesDataBaseSource.open();
        Movie movie = mMoviesDataBaseSource.getMovie(getCurrentMovieIndex());
        mMoviesDataBaseSource.close();
        return movie;
    }

    public void setMovies(List<Movie> movies) {
        mMoviesDataBaseSource.open();
        mMoviesDataBaseSource.clearMovies();
        for (int x = 0; x < movies.size(); x++)
            mMoviesDataBaseSource.insertMovie(x, movies.get(x));
        mMoviesDataBaseSource.close();
    }

}

File: app/src/main/java/com/vsa/filmoteca/data/source/repository/MoviesPersistanceRepository.kt
package com.vsa.filmoteca.data.source.repository

import com.vsa.filmoteca.domain.model.Movie
import com.vsa.filmoteca.data.source.database.MoviesDataBaseSource
import com.vsa.filmoteca.data.source.sharedpreferences.SharedPreferencesManager

import javax.inject.Inject

/**
 * Created by albertovecinasanchez on 22/7/16.
 */

class MoviesPersistanceRepository @Inject constructor(private val moviesDataBase: MoviesDataBaseSource,
                                                      private val sharedPreferencesManager: SharedPreferencesManager) {

    var currentMovieIndex: Int
        get() = sharedPreferencesManager.currentMovieIndex
        set(index) {
            sharedPreferencesManager.currentMovieIndex = index
        }

    var moviesCount: Int
        get() = sharedPreferencesManager.moviesCount
        set(size) {
            sharedPreferencesManager.moviesCount = size
        }

    val currentMovie: Movie
        get() {
            moviesDataBase.open()
            val movie = moviesDataBase.getMovie(currentMovieIndex)
            moviesDataBase.close()
            return movie
        }

    fun setMovies(movies: List<Movie>) {
        moviesDataBase.open()
        moviesDataBase.clearMovies()
        for (x in movies.indices)
            moviesDataBase.insertMovie(x, movies[x])
        moviesDataBase.close()
    }

}
